simulator/src/main/cpp/ChirpBluetoothPacket.cpp

```diff
@@ -0,0 +1,111 @@
+#include "ChirpBluetoothPacket.h"
+#include "utils.h"
+#include <string>
+
+ChirpBluetoothPacket::ChirpBluetoothPacket()
+{
+}
+
+ChirpBluetoothPacket::ChirpBluetoothPacket(char packet[], size_t size)
+{
+    this->packetType = parse_int(packet[0]);
+    this->groupId = parse_int(packet[1]);
+    
+    char node_id[] = {packet[2], packet[3]};
+    this->nodeId = parse_2digit_int(node_id);
+
+    if(this->packetType == BLUETOOTH_DATA_PACKET)
+    {
+        int i = 4;
+        int position = 0;
+        while(position < MAX_BLUETOOTH_DATA_SIZE)
+        {
+            if(i<size)
+            {
+                this->data[position] = packet[i];
+            }
+            else
+            {
+                this->data[position] = NULL;
+            }
+            i++;
+            position++;
+        }
+    }
+}
+
+int ChirpBluetoothPacket::getPacketType()
+{
+    return this->packetType;
+}
+
+void ChirpBluetoothPacket::setPacketType(int packetType)
+{
+    this->packetType = packetType;
+}
+
+bool ChirpBluetoothPacket::isInitPacket()
+{
+    return (this->packetType == BLUETOOTH_INIT_PACKET);
+}
+
+int ChirpBluetoothPacket::getGroupId()
+{
+    return this->groupId;
+}
+
+void ChirpBluetoothPacket::setGroupId(int groupId)
+{
+    this->groupId = groupId;
+}
+
+int ChirpBluetoothPacket::getNodeId()
+{
+    return this->nodeId;
+}
+
+void ChirpBluetoothPacket::setNodeId(int nodeId)
+{
+    this->nodeId = nodeId;
+}
+
+char* ChirpBluetoothPacket::getData()
+{
+    if(this->packetType == BLUETOOTH_DATA_PACKET)
+    {
+        return this->data;
+    }
+    return nullptr;
+}
+
+void ChirpBluetoothPacket::setData(char* data)
+{
+    if(this->packetType == BLUETOOTH_DATA_PACKET)
+    {
+        int i = 0;
+        while(i < MAX_BLUETOOTH_DATA_SIZE)
+        {
+            this->data[i] = data[i];
+            i++;
+        }
+    }
+}
+
+char* ChirpBluetoothPacket::serialize()
+{
+    char* packet = new char[128];
```

Comment:
where does the 128 come from?

Reply:
This comes from the Max ble packet size. ill define a global const